branchLayoutImpl/src/main/java/com/virtuslab/branchlayout/impl/manager/BranchLayoutManager.java

```diff
@@ -0,0 +1,61 @@
+package com.virtuslab.branchlayout.impl.manager;
+
+import java.nio.file.Path;
+
+import io.vavr.collection.List;
+import io.vavr.control.Try;
+import lombok.Getter;
+
+import com.virtuslab.branchlayout.api.manager.IBranchLayoutManager;
+import com.virtuslab.branchlayout.api.manager.IBranchLayoutReader;
+import com.virtuslab.branchlayout.api.manager.IBranchLayoutWriter;
+import com.virtuslab.logger.IPrefixedLambdaLogger;
+import com.virtuslab.logger.PrefixedLambdaLoggerFactory;
+
+public class BranchLayoutManager implements IBranchLayoutManager {
+  private static final IPrefixedLambdaLogger LOG = PrefixedLambdaLoggerFactory.getLogger("branchLayout");
+
+  @Getter
+  private final IBranchLayoutReader reader;
+  @Getter
+  private final IBranchLayoutWriter writer;
+
+  public BranchLayoutManager(Path path, boolean isBranchLayoutPresent) {
```

Comment:
Hmm can't just the c'tor itself decide whether the to derive indent spec or not, based on whether file under `path` exists?